extension/src/index.ts

```diff
@@ -24,4 +27,6 @@ function initializeTelemetry(): void {
 }
 
 // eslint-disable-next-line @typescript-eslint/no-empty-function
-export function deactivate(): void {}
+export async function deactivate(): Promise<void> {
+    await extension?.deactivate();
```

Comment:
infinite loop?

Reply:
here is calling function `deactivate()` from an instance of **Extension**, which will be initialized in function `activate()`